### PR TITLE
Added server configuration, added webpack configuration, implemented …

### PR DESCRIPTION
**Pull request by yarostbaklajana**
The purpose of this pull request is: 
- to add server for app
- to add webpack config for app
- to add implementation of outputting a message from the browser input  to the terminal via Socket.io

### CODE REVIEW

File: .idea/vcs.xml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (DarkXaHTeP) [inline .idea/vcs.xml:1]: This file is in .idea folder. It should be removed and ignored

File: app.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
+var express = require('express');
+var path = require('path');
+var logger = require('morgan');
+var cookieParser = require('cookie-parser');
```
Comment (DarkXaHTeP) [inline app.js]: There will be no cookies in the app

File: app.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+var bodyParser = require('body-parser');
```
Comment (DarkXaHTeP) [inline app.js]: There will be no need to parse POST requests because no POST requests will be made

File: views/layout.pug [hunk 4]
```diff
@@ -0,0 +1,8 @@
+doctype html
+html
+  head
+    title= title
+    link(rel='stylesheet', href='/stylesheets/style.css')
+    script(type="module" src='/bundle.js')
```
Comment (DarkXaHTeP) [inline views/layout.pug]: this script is NOT type module

File: public/javascripts/app.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import io from 'socket.io-client';
```
Comment (DarkXaHTeP) [inline public/javascripts/app.js]: This file should not be available to browser through `/javascripts/app.js`

File: public/bundle.js [hunk 6]
```diff
@@ -0,0 +1,6720 @@
+/******/ (function(modules) { // webpackBootstrap
```
Comment (DarkXaHTeP) [inline public/bundle.js]: This file should be removed and ignored

File: routes/index.js [hunk 7]
```diff
@@ -0,0 +1,9 @@
+var express = require('express');
```
Comment (DarkXaHTeP) [inline routes/index.js]: The app will have a single route, no need to introduce separare file for it

File: bin/www [hunk 8]
```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env node
```
Comment (DarkXaHTeP) [inline bin/www]: The server file is called `www`, it is somewhere in bin folder and imports `app` that is 2 levels up from it. [Useful information here](https://i.ytimg.com/vi/jPLKXEUm0jE/maxresdefault.jpg)

File: bin/www [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+var app = require('../app');
+var debug = require('debug')('yarostchat:server');
+var http = require('http');
+
+
+/**
+ * Get port from environment and store in Express.
+ */
+
+var port = normalizePort(process.env.PORT || '3000');
```
Comment (DarkXaHTeP) [inline bin/www]: Is it an expected behavior that your "port" variable can be either a number or a boolean?

File: bin/www (hunk 8 above)
Comment (DarkXaHTeP) [inline bin/www]: When we talked about using any templates instead of writing server from scratch I mentioned that I'll ask about some things. Please explain why do you need `#!/usr/bin/env node` at the top of the file?

File: bin/www [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+var io = require('socket.io')(server);
+
+/**
+ * Listen on provided port, on all network interfaces.
+ */
+
+server.listen(port);
+server.on('error', onError);
+server.on('listening', onListening);
+io.on('connection', function(socket){
```
Comment (DarkXaHTeP) [inline bin/www]: I guess, this part is not related to the app bootstraping and should be separated into file. Eventually there will be more business logic here

**Review by DarkXaHTeP — CHANGES_REQUESTED** [review]
Overall I would recommend next structure:
`PROJECT_ROOT/client` - folder that contains all frontend related stuff like styles, js, page template etc. All this is handled by webpack and placed into `PROJECT_ROOT//public` (this folder should be ignored in VCS)
`PROJECT_ROOT/server` - folder that contains all server related things like "server.js" - entry file that calls functions from next: "webapp.js" - express setup (BTW the only thing that we need from express is static files to serve *.js, *.html and *.css. Everything else is redundant. Keep it simple), "socket.js" - contains all listeners for socket and related logic

File: public/javascripts/app.js (hunk 5 above)
Comment (yarostbaklajana) [reply public/javascripts/app.js]: A client library that loads on the browser side: socket.io-client 
How can I use socket client other way??

Comment (DarkXaHTeP) [reply public/javascripts/app.js]: As discussed in private messaging, this comment is about `this file`, not about socket.io

File: server/socket.js [hunk 11]
```diff
@@ -0,0 +1,11 @@
+const socket = require('socket.io');
+
+module.exports = function configureSocket(server) {
+  const io = socket(server);
+
+  io.on('connection', function(socket){
+    socket.on('chat message', function(msg){
+      io.emit('chat message', `Echo: ${msg}`);
```
Comment (DarkXaHTeP) [inline server/socket.js]: Currently server sends received message to all clients, but instead it should reply to the sender only. Steps to reproduce:
1. Start server
2. Open 2 browser windows
3. Send message from one of the windows
4. Observe this message in another browser window

File: webpack.config.js [hunk 12]
```diff
@@ -1,9 +1,28 @@
 var path = require('path');
+var HtmlWebpackPlugin = require('html-webpack-plugin');
+const ExtractTextPlugin = require("extract-text-webpack-plugin");
 
 module.exports = {
-  entry: "./public/javascripts/app.js",
+  entry: "./client/app.js",
```
Comment (DarkXaHTeP) [inline webpack.config.js]: it is better to specify `context: path.join(__dirname, "client")` here and use path relative to it for entry

File: webpack.config.js [hunk 13]
```diff
@@ -1,9 +1,28 @@
...
+        use: ExtractTextPlugin.extract({
+          fallback: "style-loader",
+          use: "css-loader"
+        })
+      }
+    ]
+  },
+  plugins: [
+      new HtmlWebpackPlugin({
+        template: './client/index.html'
```
Comment (DarkXaHTeP) [inline webpack.config.js]: Could be `./index.html` if mentioned above context property is added

File: server/socket.js (hunk 11 above)
Comment (yarostbaklajana) [reply server/socket.js]: everithing is ok
